### PR TITLE
Added empty huts counter

### PR DESCRIPTION
**Pull request by MrGrim48**
Feature request #1470 

# Changes proposed in this pull request:
- Added empty huts counter

EmptyHutsCounter

This is my first contribution to another repo. 
Correct me if I may have used incorrect branches, formatting, etc.

Review please

### CODE REVIEW

**Comment by Raycoms** [comment]
Nice work, I noticed that the formatting is a bit off, you can find in the documention folder in the repo (which you downloaded) a formatter you should be able to apply to intellij or eclipse to fix that easily.

For the language file you can add the string in the en_us file to finish the todo

**Comment by MrGrim48** [comment]
I've modified the language file to accommodate the max job value and removed both todo.
This code is going on the assumption that the building name is the same as the job name as I am not sure how I can get the job names.

In any case the max jobs value should show as long as the language supports it.
Thinking about it now, a mismatch with the job/building name may show a null value.

**Comment by Raycoms** [comment]
:shipit: 

@OrionDevelopment 

[Approved with PullApprove](https://pullapprove.com/ldtteam/minecolonies/pull-request/4005/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ldtteam/minecolonies)

File: src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java [hunk 1]
```diff
@@ -635,6 +635,23 @@ private void createAndSetStatistics()
...
             }
+		}
+
+        final Map<String, Integer> jobMaxCountMap = new HashMap<>();
+        for (@NotNull final IBuildingView building : townHall.getColony().getBuildings())
+        {
+            if (!building.isBuilding())
+            {
+                String buildingName = building.getSchematicName().toLowerCase();
+                if (!buildingName.equals("warehouse") && !buildingName.equals("barracks") && !buildingName.equals("citizen") && !buildingName.equals("townhall"))
```
Comment (Asherslab) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java]: I'd much rather here you do something akin to `building instanceOf BuildingWarehouse` or etc, i think string references in this case are a lesser alternative

**Review by Asherslab — CHANGES_REQUESTED** [review]
Looks good, not sure about these string comparisons though. Instanceof usage will always work whereas a name string comparison may break if a name or etc is changed in the future

File: src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java (hunk 1 above)
Comment (marchermans) [reply src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java]: Yep!

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java]: Would be BuildWarehouse.View probably in this case though (as a note)

Comment (someaddons) [reply src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java]: instanceof AbstractBuildingWorker.View might make this even simpler

**Comment by MrGrim48** [comment]
A result of job names differing from building names. 
I've noticed a mixed use of "Baker" and "Bakery".

doubleBaker

**Comment by Raycoms** [comment]
We could check if the job is contained in the building for this @MrGrim48

**Comment by Raycoms** [comment]
@MrGrim48 I thought about it.
In the Building classes we already jave a getJobName(), if we simply add this method to the view as well. We avoid any issues and can use it consistently for all buildings.

    @NotNull
    @Override
    public String getJobName()

**Comment by MrGrim48** [comment]
Initially I wanted to use the method `getJobName()` but didn't directly have access to it.
From what I can see with the baker, that method would return the same string as the schematic name.

The bakers job returns the string constant `COM_MINECOLONIES_COREMOD_JOB_BAKER `from `TranslationConstants.java` which returns bakery instead of baker.

From my understanding, the buildings create the jobs and pass it to the citizen to hold onto and if that is the case, I don't know how I could obtain the jobs name in the event that no citizens have the job.

Even if we have `getJobName()` return the job constant instead, I feel that adding the method to the view is beyond my current understanding.
The general use of abstraction and interfaces alone is currently foreign to me with which I'm still trying to get used to.

**Comment by Raycoms** [comment]
Okay, sounds good, this should work for most of the cases for now.
:shipit:

**Comment by Asherslab** [comment]
My review still stands

**Comment by Raycoms** [comment]
Yeah, he adapted it now to instanceof

**Review by Asherslab — APPROVED** [review]
If it’s tested and working, looks good to go

**Pull request merged**